src/GraphQL/Field/Root/Entity/EntityQueryField.php

```diff
@@ -144,16 +170,20 @@ public function resolve($parent, array $args = NULL, ResolveInfo $info) {
     }
 
     $result = $query->execute();
-    if (!empty($result)) {
+    if (empty($result)) {
+      $filteredEntities = [];
+    }
+    else {
       $entities = $storage->loadMultiple($result);
 
-      // Filter entities that the current user doesn't have view access for.
-      return array_filter($entities, function (EntityInterface $entity) {
-        return $entity->access('view');
-      });
+      $filteredEntities = array_filter($entities,
+        function (EntityInterface $entity) {
+          return $entity->access('view');
+        }
+      );
     }
 
-    return [];
+    return ArrayConnection::connectionFromArray($filteredEntities, $args);
```

Comment:
This is fine as an interim solution but we need to do the Connection logic on the query level to make this feasible for production.